JamJar/VideoPageViewController.swift

```diff
@@ -94,10 +110,142 @@ class VideoPageViewController: BaseViewController, UITableViewDelegate, UITableV
         self.view.layoutIfNeeded()
     }
     
+    func removeOberservsInPlayer() {
+        for controller in self.childViewControllers {
+            if let child = controller as? JamJarAVPlayerViewController {
+                child.removeObservers()
+            }
+        }
+    }
+    
+    func changeVideoInPlayer() {
+        for controller in self.childViewControllers {
+            if let child = controller as? JamJarAVPlayerViewController {
+                child.removeObservers()
+                let videoPath = NSURL(string: self.video.hls_src)
+                child.player = AVPlayer(URL: videoPath!)
+                child.viewDidLoad()
+            }
+        }
+    }
+    
+    func updateLikeDislikeButtons() {
+        self.likeButton.selected = (video.userVote == true)
+        self.dislikeButton.selected = (video.userVote == false)
+        self.likeButton.highlightBackgroundLike()
+        self.dislikeButton.highlightBackgroundDislike()
+    }
+    
     func concertTapped(sender:AnyObject) {
         self.performSegueWithIdentifier("ToConcertPage", sender: nil)
     }
     
+    @IBAction func likePressed(sender: UIButton) {
+        if(video.userVote == true) {
+            VideoService.vote(video.id, vote: nil) { success, error in
+                if !success {
+                    // Error - show the user
+                    let errorTitle = "Video error!"
+                    if let error = error { SCLAlertView().showError(errorTitle, subTitle: error, closeButtonTitle: "Got it") }
+                    else { SCLAlertView().showError(errorTitle, subTitle: "", closeButtonTitle: "Got it") }
+                } else {
+                    self.video.userVote = nil
+                    self.updateLikeDislikeButtons()
+                }
+            }
+        } else {
+            VideoService.vote(video.id, vote: true) { success, error in
```

Comment:
Alright soooo, we have a whole bunch of code cloned here as well.  Maybe make a base player page class or something?